src/wui/interactive_base.cc

```diff
@@ -708,6 +709,18 @@ void InteractiveBase::game_logic_think() {
 	egbase().think();  // Call game logic here. The game advances.
 }
 
+void InteractiveBase::think() {
+	UI::Panel::think();
+	if (in_road_building_mode()) {
+		set_tooltip(
+		   in_road_building_mode(RoadBuildingType::kRoad) ?
+		      (boost::format(_("Road length: %u")) % get_build_road_path().get_nsteps()).str() :
```

Comment:
Please extract that ? : expressssion, together wth that boost format with % I almost got lost